backend/routes/meta.js

```diff
@@ -273,14 +361,43 @@
 			});
 	}
 
+	function queryMetadataByRules(req, res, next) {
+		const rules = req.body;
+		let branch;
+
+		if (!req.params.rev) {
+			branch = C.MASTER_BRANCH_NAME;
+		}
+
+		const showMeshIds = (req.query.meshids) ? JSON.parse(req.query.meshids) : false;
+
+		if (showMeshIds) {
```

Comment:
less duplication if we do this 
```suggestion
                const promise = showMeshIds ? Meta.getMeshIdsByRules(req.params.account, req.params.model, branch, req.params.rev, rules) : Meta.getAllMetadataByRules(req.params.account, req.params.model, branch, req.params.rev, rules);
                
                promise.then((obj) => ...)

```